### PR TITLE
PUSH OF MINOR INTO MAJOR

### PR DESCRIPTION
**Pull request by ridz1208**

### CODE REVIEW

**Comment by ridz1208** [comment]
@johnsaigle I had a few conflicts in this one, could you do a sanity check on the changes ?

File: modules/data_release/php/data_release.class.inc [hunk 1]
```diff
@@ -72,41 +72,133 @@ class Data_Release extends \NDB_Menu_Filter
...
+     * Gets the array of release versions with their corresponding files.
+     *
+     * This function is used for the initial load as well as for comparison upon
+     * update to ensure no values are added or deleted unexpectedly.
+     *
+     * @param \Database $DB Database to use for querying
+     *
+     * @return array Associative array of $versions=>(files associate with version)
+     */
+    function getVersionedFiles(\Database $DB)
```
Comment (johnsaigle) [inline modules/data_release/php/data_release.class.inc]:
```suggestion
    function getVersionedFiles(\Database $DB): array
```

File: modules/imaging_uploader/ajax/getUploadSummary.php [hunk 2]
```diff
@@ -17,48 +17,22 @@
...
-    $summary = $_POST['summary'] == 'true';
-}
+$uploadId = $_POST['uploadId'];
+$summary  = $_POST['summary'] === 'true';
 
-// Fetch columns Inserting and InsertionComplete from table mri_upload
+/* Fetch columns Inserting and InsertionComplete from table mri_upload
+ * create Database object
+ */
+$DB    =& Database::singleton();
```
Comment (johnsaigle) [inline modules/imaging_uploader/ajax/getUploadSummary.php]:
```suggestion
$DB    = \Database::singleton();
```

File: modules/media/ajax/FileUpload.php [hunk 3]
```diff
@@ -196,35 +196,35 @@ function viewData()
...
-    $sessionRecords = $db->pselect(
-        "SELECT c.PSCID, s.Visit_label, s.CenterID, f.Test_name " .
-        "FROM candidate c ".
-        "LEFT JOIN session s USING(CandID) ".
-        "LEFT JOIN flag f ON (s.ID=f.SessionID) ".
-        "ORDER BY c.PSCID ASC",
-        []
-    );
-
+    $sessionData = [];
```
Comment (johnsaigle) [inline modules/media/ajax/FileUpload.php]:
```suggestion
    $sessionData = array();
```

**Comment by ridz1208** [comment]
@johnsaigle are these suggestions additions or just bad rebase ?

if they are not due to bad resolution of conflicts I dont want to commit them here

**Comment by johnsaigle** [comment]
Additions. Don't add them if you're feeling cautious.

**Comment by ridz1208** [comment]
@johnsaigle created issues for these.

merging the PR

**Pull request merged**